cpp/src/rolling/rolling.cu

```diff
@@ -136,9 +237,49 @@ template <typename InputType>
 struct rolling_window_launcher
 {
 
+  template <typename T, typename agg_op, aggregation::Kind op, typename WindowIterator, bool op_argmin_agrmax=false>
+  void kernel_launcher(column_view const& input,
+                       mutable_column_view& output,
+                       WindowIterator preceding_window_begin,
+                       WindowIterator following_window_begin,
+                       size_type min_periods,
+                       std::unique_ptr<aggregation> const& aggr,
+                       T identity,
+                       cudaStream_t stream) {
+      
+      cudf::nvtx::range_push("CUDF_ROLLING_WINDOW", cudf::nvtx::color::ORANGE);
+
+      constexpr cudf::size_type block_size = 256;
+      cudf::experimental::detail::grid_1d grid(input.size(), block_size);
+
+      auto input_device_view = column_device_view::create(input, stream);
+      auto output_device_view = mutable_column_device_view::create(output, stream);
+
+      rmm::device_scalar<size_type> device_valid_count{0, stream};
+
+      if (input.has_nulls()) {
+          gpu_rolling<T, target_type_t<InputType, op>, agg_op, op, block_size, op_argmin_agrmax, true><<<grid.num_blocks, block_size, 0, stream>>>
+              (*input_device_view, *output_device_view, device_valid_count.data(),
+               preceding_window_begin, following_window_begin, min_periods, identity);
+      } else {
+          gpu_rolling<T, target_type_t<InputType, op>, agg_op, op, block_size, op_argmin_agrmax, false><<<grid.num_blocks, block_size, 0, stream>>>
+              (*input_device_view, *output_device_view, device_valid_count.data(),
+               preceding_window_begin, following_window_begin, min_periods, identity);
+      }
+
+      output.set_null_count(output.size() - device_valid_count.value(stream));
+
+      // check the stream for debugging
+      CHECK_CUDA(stream);
+      
+      cudf::nvtx::range_pop();
+
+  }
+
   template <typename T, typename agg_op, aggregation::Kind op, typename WindowIterator>
-  std::enable_if_t<cudf::detail::is_supported<T, agg_op,
-                                  op, op == aggregation::MEAN>(), std::unique_ptr<column>>
+  std::enable_if_t<(cudf::detail::is_supported<T, agg_op,
```

Comment:
There are several SFINAE specifializations for launch now, and I find them hard to track. Can you add a comment to each one saying which cases it handles? It will make future support easier.